### ISSUE TITLE
UHK mouse over bluetooth is significantly laggier than my other regular bluetooth mouse

### ISSUE THREAD

**Issue by mhantsch** [UltimateHackingKeyboard/firmware#1025]
I've used bluetooth mice a lot in the past years, so I have experience in what I can expect in terms of latency and responsiveness. But maybe they don't use bluetooth LE; don't know.

When I use the UHK 80 over bluetooth, the mouse reaction seems much more sluggish than with a regular bluetooth mouse. It doesn't matter whether I use the trackpad ore the trackball on the UHK.

When I plug in the UHK 80 and use it via USB, there is no sluggishness, no latency.

**Comment by kareltucek** [comment]
You mean latency, speed, or movement smoothness?

**Comment by mhantsch** [comment]
I mean latency. It feels more sluggish. My bluetooth mouse feels more responsive, more direct.

**Comment by kareltucek** [comment]
So the movement is smooth and fast, but you feel like there is an input lag. Is that correct?

**Comment by mhantsch** [comment]
Yes, once it is moving it is moving, but everything is kind of a little bit delayed compared to another bluetooth mouse. So, by using the other mouse I know that this can be more responsive over bluetooth. BT is not the limiting factor, at least not in principle. It's something else. Maybe the way the UHK uses the BT protocols and stack, or it's somewhere else before it reaches the BT stack.

Another chance is that because my left half doesn't want to talk wirelessly to the right half, something else is messed up which uses BT ressources, and that causes the mouse lag. I'll open another bug report for the left-right halves connection issue.

**Comment by mhantsch** [comment]
Interconnect issue between halves followed up in https://github.com/UltimateHackingKeyboard/firmware/issues/970.

**Comment by kareltucek** [comment]
@mhantsch I have optimized bluetooth settings recently. Can you give it a test?

**Comment by mhantsch** [comment]
> [@mhantsch](https://github.com/mhantsch) I have optimized bluetooth settings recently. Can you give it a test?

I think it's much better now. I don't sense "significantly laggier" behaviour anymore. I think we can close this one.
